internal/conf/metrics.go

```diff
@@ -0,0 +1,192 @@
+		},
+	})
+
+	return client, nil
+}
+
+type memoryReporter struct {
+	statsd statsd.ClientInterface
+	logger *zap.SugaredLogger
+}
+
+func NewMemoryReporter(lc fx.Lifecycle, statsd statsd.ClientInterface, logger *zap.SugaredLogger) *memoryReporter {
+	mr := &memoryReporter{
+		statsd: statsd,
+		logger: logger,
+	}
+	lc.Append(fx.Hook{
+		OnStart: func(ctx context.Context) error {
+			mr.init()
+			return nil
+		},
+	})
+
+	return mr
+}
+
+func (mr *memoryReporter) init() {
+	descs := metrics.All()
+	// Create a sample for each metric.
+	samples := make([]metrics.Sample, len(descs))
+	for i := range samples {
+		samples[i].Name = descs[i].Name
+	}
+
+	go func() {
+		for {
+			time.Sleep(15 * time.Second)
+			mr.run(samples)
+		}
+	}()
+
+}
+
+func (mr *memoryReporter) run(samples []metrics.Sample) {
+
+	// Sample the metrics. Re-use the samples slice if you can!
+	metrics.Read(samples)
+
+	// Iterate over all results.
+	for _, sample := range samples {
+		// Pull out the name and value.
+		name, value := strings.ReplaceAll(strings.ReplaceAll(sample.Name, "/", ".")[1:], ":", "."), sample.Value
+
+		// Handle each sample.
+		switch value.Kind() {
+		case metrics.KindUint64:
+			_ = mr.statsd.Gauge(name, float64(value.Uint64()), nil, 1)
+		case metrics.KindFloat64:
+			_ = mr.statsd.Gauge(name, value.Float64(), nil, 1)
+		case metrics.KindFloat64Histogram:
+			// The histogram may be quite large, so let's just pull out
+			// a crude estimate for the median for the sake of this example.
+			//val := medianBucket(value.Float64Histogram())
+			//_ = mr.statsd.Histogram(name+".hist", val, nil, 1)
+			//_ = mr.statsd.Gauge(name, val, nil, 1)
+			val := getHistogram(value.Float64Histogram())
+			_ = mr.statsd.Gauge(name+".avg", val.avg, nil, 1)
+			_ = mr.statsd.Count(name+".count", val.count, nil, 1)
+			_ = mr.statsd.Gauge(name+".median", val.median, nil, 1)
+			_ = mr.statsd.Gauge(name+".max", val.max, nil, 1)
+		case metrics.KindBad:
+			// This should never happen because all metrics are supported
+			// by construction.
+			mr.logger.Error("bug in runtime/metrics package!")
+		default:
+			// This may happen as new metrics get added.
+			//
+			// The safest thing to do here is to simply log it somewhere
+			// as something to look into, but ignore it for now.
+			// In the worst case, you might temporarily miss out on a new metric.
+			mr.logger.Warnf("%s: unexpected metric Kind: %v\n", name, value.Kind())
+		}
+	}
+
+	// still support old metrics until new is in place+
+	var m runtime.MemStats
+	runtime.ReadMemStats(&m)
+
+	_ = mr.statsd.Gauge("mem.alloc", float64(bToMb(m.Alloc)), nil, 1)
+	_ = mr.statsd.Gauge("mem.totalAlloc", float64(bToMb(m.TotalAlloc)), nil, 1)
+	_ = mr.statsd.Gauge("mem.sys", float64(bToMb(m.Sys)), nil, 1)
+	_ = mr.statsd.Gauge("heap.obj", float64(m.Mallocs-m.Frees), nil, 1)
+	_ = mr.statsd.Gauge("heap.sys", float64(bToMb(m.HeapSys)), nil, 1)
+}
+
+func bToMb(b uint64) uint64 {
+	return b / 1024 / 1024
```

Comment:
return float64 to
a) keep precision. `float64(3100000) / 1024 / 1024` returns round 2.95. `3100000 / 1024 / 1024` returns 2
b) avoid casting when building gague